### PR TITLE
feat: Upgrade to pywr-v1-schema v0.8.0.

### PR DESCRIPTION
**Pull request by jetuk**
Required implementing several new parameters:

- InterpolatedParameter
- DiscountFactorParameter
- DataFrameParameter

Conversion from Storage, Deficit and Flow parameters is now returns an error as these should become metrics in pywr-next.

Todos are left for HydropowerTarget, RollingMeanFlowNode and ScenarioWrapper parameter.

Also the control curve interpolated parameter was renamed due to a naming conflict with InterpolatedParameter.

### CODE REVIEW

**Review by Batch21 — CHANGES_REQUESTED** [review]
Couple of queries but otherwise this looks good

File: pywr-schema/src/data_tables/scalar.rs [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+
+            if values.len() > 1 {
+                return Err(TableError::TooManyValues(path.clone()));
+            }
+
+            Ok((key, values[0]))
+        })
+        .collect::<Result<_, TableError>>()?;
+
+    Ok(LoadedScalarTable::Three(ScalarTableThree { values }))
```
[the remark this answers is not in the document]
Comment (jetuk) [reply pywr-schema/src/data_tables/scalar.rs:241]: Hmm yes. Not sure how this is compiling!

**Comment by jetuk** [comment]
This was a bit broken. I've fixed the `data_tables` module. At least to the extent it has the same functionality. It could do with completing other cases and having some extensive tests & documentation.

**Review by Batch21 — APPROVED** [review]
fixes look good

**Pull request merged**